File: src/main/java/net/sf/rails/game/model/PresidentModel.java
package net.sf.rails.game.model;

import net.sf.rails.game.Player;
import net.sf.rails.game.PublicCompany;

/**
 * model object for the current company president
 * gets registered by the ShareModels
 * 
 * FIXME: Finalize implementation, this does not work currently
 * TODO: Check if this is all done correctly, where is the observable stored?
 */

public final class PresidentModel extends RailsModel {

    public static final String ID = "PresidentModel";  
    
    private PresidentModel(PublicCompany parent, String id) {
        super(parent, id);
    }

    public static PresidentModel create(PublicCompany parent){
        return new PresidentModel(parent, ID);
    }
    
    /**
     * @return restricted to PublicCompany
     */
    @Override
    public PublicCompany getParent() {
        return (PublicCompany)super.getParent();
    }

    @Override
    public String toText() {
        Player president = getParent().getPresident();
        if (president == null) return "";
        else return getParent().getPresident().getNameAndPriority();
    }

}

File: dotnet/RailsLib.Net/Net/Game/Model/PresidentModel.cs
using System;
using System.Collections.Generic;
using System.Text;

/**
 * model object for the current company president
 * gets registered by the ShareModels
 * 
 * FIXME: Finalize implementation, this does not work currently
 * TODO: Check if this is all done correctly, where is the observable stored?
 */

namespace GameLib.Net.Game.Model
{
    public class PresidentModel : RailsModel
    {
        public const string ID = "PresidentModel";

        private PresidentModel(PublicCompany parent, string id) : base(parent, id)
        {

        }

        public static PresidentModel Create(PublicCompany parent)
        {
            return new PresidentModel(parent, ID);
        }

        /**
         * @return restricted to PublicCompany
         */
        new public PublicCompany Parent
        {
            get
            {
                return (PublicCompany)base.Parent;
            }
        }

        override public string ToText()
        {
            Player president = Parent.GetPresident();
            if (president == null) return "";
            else return Parent.GetPresident().GetNameAndPriority();
        }
    }
}
